components/banners.js

```diff
@@ -0,0 +1,79 @@
+import Alert from 'react-bootstrap/Alert'
+import styles from './banners.module.css'
+import { useEffect, useState } from 'react'
+import { useMe } from '../components/me'
+import { useMutation } from '@apollo/client'
+import { WELCOME_BANNER_MUTATION } from '../fragments/users'
+
+export default function NewVisitorBanner () {
+  const today = new Date()
+  const sameDay = (a, b) => {
+    return Math.abs(a.getTime() - b.getTime()) < 1000 * 24 * 60 * 60
+  }
+  const me = useMe()
+  const [state, setState] = useState(null)
+  const handleClose = async () => {
+    const obj = { date: state?.date || new Date().toJSON(), hidden: true }
+    const str = JSON.stringify(obj)
+    window.localStorage.setItem('newVisitorInfo', str)
+    setState(obj)
+    const { error } = await hideWelcomeBanner({ variables: { } })
+    if (error) {
+      throw new Error({ message: error.toString() })
```

Comment:
This throws an error if you are not logged in.

This mutation shouldn't run if you are not logged in (to prevent unnecessary errors) and if you are logged in, it should use our new [toast](https://github.com/stackernews/stacker.news/pull/431) mechanism or something else to tell the user an error happened.

Reply:
Perfect! I'll try to get that in today.